internal/pkg/service/buffer/file/manager.go

```diff
@@ -0,0 +1,50 @@
+package file
+
+import (
+	"net/http"
+
+	"github.com/benbjohnson/clock"
+	"github.com/keboola/go-client/pkg/keboola"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/task"
+)
+
+const (
+	gzipLevel = 2 // 1 - BestSpeed, 9 - BestCompression
+)
+
+type Manager struct {
+	clock     clock.Clock
+	store     *store.Store
+	tasks     *task.Node
+	publicAPI *keboola.API
+	transport http.RoundTripper
+}
+
+type manager = Manager
+
+type AuthorizedManager struct {
+	*manager
+	projectAPI *keboola.API
+}
+
+type dependencies interface {
+	Clock() clock.Clock
+	Store() *store.Store
+	KeboolaPublicAPI() *keboola.API
+	TaskNode() *task.Node
+}
+
+func NewManager(d dependencies) *Manager {
+	return &Manager{
+		clock:     d.Clock(),
+		publicAPI: d.KeboolaPublicAPI(),
+		store:     d.Store(),
+		tasks:     d.TaskNode(),
+	}
+}
+
+func (m *Manager) WithToken(token string) *AuthorizedManager {
+	return &AuthorizedManager{manager: m, projectAPI: m.publicAPI.WithToken(token)}
+}
```

Comment:
Tu bol problem s autorizaciou.

Na viacerych miestach v kode sa vytvarala instancia API s tokenom.
V API operaciach to bol token request, vo worker operacia to bol token so scoce iba na bucket.

Teraz som to zjednotil, a vsade to bude token so scope na bucket, budeme to potrebovat aj pri SOX.

Okrem toho sa autorizovana instancia API vytvara iba na jednom mieste, tu.